### DISCUSSION TITLE
How to add other currencies

### DISCUSSION THREAD

**Discussion by mikeabiz** [inventree/InvenTree#8080] [Q&A] [answered]
### Body of the issue

I am trying to find documentation on how to exactly add other currencies, example MYR, PHP, etc.  I am only seeing the several default currencies and could not figure how to add the others.

**Comment by mikeabiz** [comment]
Not sure how to proceed changing the currency from USD to MYR or PHP.  Any clear documentation would be most appreciated

**Reply by tristanle22** [reply]
I'm having the same issue. I want to add currency VND.

**Reply by matmair** [reply]
Thank you for reading the docs / using google https://docs.inventree.org/en/stable/settings/currency/

**Comment by mikelemo** [comment]
Well it's going to be a bit hard to get help from the main developers enving this project or somethings but if this goes on like this we can go build another inventree with blackjack and hookrs, infact forget the tree. 
Adding the currency depends it might have been fixed since but back in the time I had to add it manually to an inventree file probably the config but now you might be able to go to (after starting inventree) to http://localhost:8000/platform/settings/system/pricing
(System settings --> Pricing) then add a currency according to "Supported Currencies" with the appropriate code django money supports: 
https://github.com/adamcharnock/django-hordak/blob/master/hordak/utilities/currency.py

```py
class FixerBackend(BaseBackend):
    """Use fixer.io for currency conversions"""

    supported_currencies = [
        "AUD",
        "BGN",
        "BRL",
        "CAD",
        "CHF",
        "CNY",
        "CZK",
        "DKK",
        "GBP",
        "HKD",
        "HRK",
        "HUF",
        "IDR",
        "ILS",
        "INR",
        "JPY",
        "KRW",
        "MXN",
        "MYR",
        "NOK",
        "NZD",
        "PHP",
        "PLN",
        "RON",
        "RUB",
        "SEK",
        "SGD",
        "THB",
        "TRY",
        "ZAR",
        "EUR",
        "USD",
    ]
```

**Comment by lpezo** [comment]
Tengo esta solución para tu moneda:

Primero, configura el Inventree
En Ajustes del Sistema / Precios / Supported Currencies; Agrega tu moneda, y borra lo que no necesites, debería detectarlo. Luego de "Default Currency" selecciona tu moneda
<img width="483" height="74" alt="image" src="https://github.com/user-attachments/assets/076b921f-1666-4b98-a172-10da8e8f019e" />

Eso debe funcionar para la mayor parte del sistema, en algunas situaciones de precios no lo detecta, has esto también:

Segundo Paso
Despues de instalarlo, busca en el servidor en el directorio instalado, abre el archivo currency.py, paso ruta:
    /src/backend/InvenTree/common/currency.py
busca la la funcion: def currency_codes_default_list() -> str:
encontrarás lo siguiente, simplemente agrega tu moneda y graba
    return 'AUD,CAD,CNY,EUR,GBP,JPY,NZD,USD,<TU MONEDA>'
Reinicia el servicio de Inventree-Server. 

Y listo.

**Reply by matmair** [reply]
Please use English in the issue tracker and discussions to reduce possible translation errors

**Reply by lpezo** [reply, marked as answer]
OK,
First:
Configure your currency in the application:
In System Settings / Pricing / Supported Currencies, add your currency code, and delete any codes you don't need.
<img width="345" height="95" alt="image" src="https://github.com/user-attachments/assets/4196cd25-0d9f-4405-8be2-c513221d4490" />
Then, in "Default Currency" select your currency.
<img width="286" height="209" alt="image" src="https://github.com/user-attachments/assets/46cb1d1b-861d-4310-b696-2693285eae1e" />

Finally: Fix the code after installation. In my case, I installed it in Docker. Run the command in the terminal:
$ sudo docker exec -it inventree-server bash

> apt-get update

> apt-get install nano

> nano src/backend/InvenTree/common/currency.py

Find "def currency_codes_default_list() -> str:"
Modify "return 'AUD,CAD,CNY,EUR,GBP,JPY,NZD,USD,XXX'"
Note: XXX is the code for your currency.
Save with Ctrl+x
Exit bash:

> exit
And restart the service. In my case, with Docker, it's:
$ sudo docker restart inventree-server

And that's it! Your currency should work.
